src/main/java/org/testng/internal/ConfigurationMethod.java

```diff
@@ -456,11 +457,15 @@ public boolean isLastTimeOnly() {
   }
 
   public String[] getGroupFilters() {
```

Comment:
I'm not overly happy with this code. I'd like to have something like
```java
IBaseBeforeAfterMethod beforeAfter = m_annotationFinder.findAnnotation(
    getConstructorOrMethod(), IBaseBeforeAfterMethod .class);
```
But unfortunately `AnnotationFinder` doesn't support base interfaces that have no corresponding annotations.

Reply:
Feel free to add the method yourself ;)

Reply:
I would, but I can't seem to find an easy and elegant way to do it. It's strongly based on a map with one-to-one correspondence. I could add another map for base classes with one-to-many correspondence, but then I'll have to go through the whole class and do something about places where that map is used. Not a trivial task, but not too simple either, just to make a single method a couple of lines shorter.

Reply:
The value could be a complex object too right ? Perhaps like a list or a custom data structure ?

Reply:
Right, of course. But that's not a problem. The problem is that is code depends too strongly on the one-to-one correspondence. And I'm not sure how to move away from that. Not only implementation-wise, but design-wise as well. For example, in this case I need to find just one annotation of either `IBeforeMethod` or `IAfterMethod` type. It doesn't make sense to use both (although in theory a user could). But what if someone calls this method asking for `IAnnotation`? Should it return the first annotation it finds or the list of all annotations maybe?

Reply:
@stachenov - Then I think we should perhaps leave it as is, as part of this PR, and perhaps deal with it in a separate PR. What do you think ?

Reply:
Yes, I definitely think so. It's not worth redesigning a substantial part of the project just to make a small utility method a bit more elegant.